amqp/amqp.go

```diff
@@ -95,38 +96,71 @@ func New(url string) (queue.Broker, error) {
 		},
 	}
 
+	b.connErrors = make(chan *amqp.Error, 1)
+	b.conn.NotifyClose(b.connErrors)
+
+	b.chErrors = make(chan *amqp.Error, 1)
+	b.ch.NotifyClose(b.chErrors)
+
 	go b.manageConnection(url)
 
 	return b, nil
 }
 
 func (b *Broker) manageConnection(url string) {
-	b.connErrors = make(chan *amqp.Error)
-	b.conn.NotifyClose(b.connErrors)
-
 	for {
 		select {
 		case err := <-b.connErrors:
-			log.Errorf(err, "amqp connection error")
-			b.mut.Lock()
-			if err != nil {
-				b.conn, b.ch = b.reconnect(url)
-				b.connErrors = make(chan *amqp.Error)
-				b.conn.NotifyClose(b.connErrors)
+			log.Errorf(err, "amqp connection error - reconnecting")
+			if err == nil {
+				break
+			}
+
+			// b.mut.Lock()
```

Comment:
Do we need those locks or can we delete the lines? Same in `b.chErrors` block.

Reply:
no, my bad - leftovers